low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -501,14 +505,77 @@ def validate(self):
+                directories = [
+                    f'/var/{PRODUCT_FAMILY}/sspl',
+                    f'/var/{PRODUCT_FAMILY}/iem/',
+                    f'/var/log/{PRODUCT_FAMILY}/sspl/',
+                    f'/var/log/{PRODUCT_FAMILY}/iem/',
+                    '/etc/sspl-ll/', f'{PRODUCT_BASE_DIR}/iem/iec_mapping']
+                sspl_sudoers_file = '/etc/sudoers.d/sspl'
+                sspl_dbus_policy_rules = '/etc/polkit-1/rules.d'
+                sspl_dbus_policy_conf = '/etc/dbus-1/system.d'
+                sspl_service_file = '/etc/systemd/system/sspl-ll.service'
+                sspl_test_backup = '/etc/sspl_tests.conf.back'
+                sspl_test_file_path = '/etc/sspl_test_gc_url.yaml'
+                sspl_sb_log_file_path = sspl_log_file_path.replace(
+                    "/sspl.log", "/sspl_support_bundle.log")
+                manifest_log_file_path = sspl_log_file_path.replace(
+                    "/sspl.log", "/manifest.log")
+                # symlinks created during post_install
+                sspl_ll_cli = "/usr/bin/sspl_ll_cli"
+                # Remove SSPL config other config files which we have created
+                # during post_install.
+                for file in [
+                    sspl_ll_cli, sspl_test_backup,
+                    sspl_test_file_path, file_store_config_path,
+                    global_config_file_path, sspl_log_file_path,
+                    iem_log_file_path, sspl_sb_log_file_path,
+                    manifest_log_file_path, RSYSLOG_IEM_CONF,
+                    RSYSLOG_SSPL_CONF, RSYSLOG_MSB_CONF,
+                    IEM_LOGROTATE_CONF, SSPL_LOGROTATE_CONF,
+                    MSB_LOGROTATE_CONF, RSYSLOG_SB_CONF,
+                    SB_LOGROTATE_CONF, sspl_dbus_policy_conf,
+                    sspl_dbus_policy_rules, sspl_sudoers_file,
+                        sspl_service_file]:
+                    self.delete_config(file)
+                # Delete directories which we have created during post_install.
+                for dir in directories:
+                    self.delete_config(dir, del_dir=True)
+                # Delete sspl-ll user
+                usernames = [x[0] for x in pwd.getpwall()]
+                if USER in usernames:
+                    os.system("/usr/sbin/userdel -f %s" % USER)
             logger.info("%s - Process done" % self.name)
         except OSError as e:
             logger.error(f"Failed in Cleanup. ERROR: {e}")
 
+    def delete_config(self, path, del_dir=False):
+        """Delete directories/files from dir which starts with 'sspl'."""
+        if not os.path.exists(path):
+            logger.info(f"{path} path doesn't exists.")
+            return
+        if os.path.isfile(path):
+            os.remove(path)
+        elif os.path.islink(path):
+            os.unlink(path)
+        elif os.path.isdir(path) and del_dir:
+            shutil.rmtree(path, ignore_errors=True)
+
```

Comment:
We can keep these common functions in Utility class.

Reply:
Done.

Reply:
No new code should be added under generic Utility class anymore. it needs to be under class for specific purpose, like file_utils in this PR